api/scpca_portal/test/models/test_library.py

```diff
@@ -0,0 +1,119 @@
+        self.assertIn(self.library1, result)
+
+    def test_get_project_libraries_from_download_config_includes_merged_no_merged_file(self):
+        config = self.default_config.copy()
+        config["includes_merged"] = True
+        config["excludes_multiplexed"] = True
+        self.project.includes_merged_sce = False
+        self.project.includes_merged_anndata = False
+
+        result = Library.get_project_libraries_from_download_config(self.project, config)
+        self.assertEqual(result.count(), 0)
+
+    def test_get_project_libraries_from_download_config_metadata_only(self):
+        config = {
+            "modality": None,
+            "format": None,
+            "excludes_multiplexed": None,
+            "includes_merged": None,
+            "metadata_only": True,
+        }
+        result = Library.get_project_libraries_from_download_config(self.project, config)
+        self.assertIn(self.library1, result)
+        self.assertIn(self.library2, result)
+        self.assertIn(self.library3, result)
+
+
+class TestGetSampleLibrariesFromDownloadConfig(TestCase):
+    def setUp(self):
+        self.sample = SampleFactory()
+        self.library1 = LibraryFactory(
+            modality=Library.Modalities.SINGLE_CELL,
+            formats=[Library.FileFormats.SINGLE_CELL_EXPERIMENT, Library.FileFormats.ANN_DATA],
+        )
+        self.library2 = LibraryFactory(
+            modality=Library.Modalities.SINGLE_CELL,
+            formats=[Library.FileFormats.SINGLE_CELL_EXPERIMENT],
+        )
+        self.library3 = LibraryFactory(
+            modality=Library.Modalities.SINGLE_CELL, formats=[Library.FileFormats.ANN_DATA]
+        )
+        self.library4 = LibraryFactory(
+            modality=Library.Modalities.SPATIAL,
+            formats=[Library.FileFormats.SINGLE_CELL_EXPERIMENT],
+        )
+        self.sample.libraries.add(self.library1, self.library2, self.library3, self.library4)
+
+    def test_get_sample_libraries_from_download_config_single_cell_sce(self):
+        config = {"modality": "SINGLE_CELL", "format": "SINGLE_CELL_EXPERIMENT"}
+        result = Library.get_sample_libraries_from_download_config(self.sample, config)
+        self.assertIn(self.library1, result)
+        self.assertIn(self.library2, result)
+
+    def test_get_sample_libraries_from_download_config_single_cell_anndata(self):
+        config = {"modality": "SINGLE_CELL", "format": "ANN_DATA"}
+        result = Library.get_sample_libraries_from_download_config(self.sample, config)
+        self.assertIn(self.library1, result)
+        self.assertIn(self.library3, result)
+
+    def test_get_sample_libraries_from_download_config_spatial_sce(self):
+        config = {"modality": "SPATIAL", "format": "SINGLE_CELL_EXPERIMENT"}
+        result = Library.get_sample_libraries_from_download_config(self.sample, config)
+        self.assertIn(self.library4, result)
```

Comment:
Lets combine these into a single test and iterate over the actual common. we have defined. I would iterate over each of the configs and then check the modality and format in the download_configuration and check that the appropriate samples are there.